From b29f0fd02ffd9facaedb99ff4a5f713a318c56e6 Mon Sep 17 00:00:00 2001
From: Algernon <lixiaodong67@huawei.com>
Date: Tue, 20 Jun 2023 10:33:29 +0800
Subject: [PATCH 053/283] net:hns3 Fix KABI for
 ethtools->supported_coalesce_params

driver inclusion
category: bugfix
bugzilla: https://gitee.com/src-openeuler/kernel/issues/I8EMQV

--------------------------------

Fix ethtool_ Added supported in the ops structure_ Coalesce_ Kabi
changes caused by params members.

Signed-off-by: Xiaodong Li <lixiaodong67@huawei.com>
---
 include/linux/ethtool.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/ethtool.h b/include/linux/ethtool.h
index 217f56d2462c..65c244c2dc97 100644
--- a/include/linux/ethtool.h
+++ b/include/linux/ethtool.h
@@ -367,7 +367,9 @@ bool ethtool_convert_link_mode_to_legacy_u32(u32 *legacy_u32,
  * of the generic netdev features interface.
  */
 struct ethtool_ops {
+#ifndef __GENKSYMS__
 	u32     supported_coalesce_params;
+#endif
 	int	(*get_settings)(struct net_device *, struct ethtool_cmd *);
 	int	(*set_settings)(struct net_device *, struct ethtool_cmd *);
 	void	(*get_drvinfo)(struct net_device *, struct ethtool_drvinfo *);
-- 
2.34.1

